### PR TITLE
use eslint JSON output over errorformat

### PR DESCRIPTION
**Pull request by euclio**
`eslint` provides a JSON output format. Using it gives us accurate highlight lengths.

### CODE REVIEW

File: tests/ft_javascript.vader [hunk 1]
```diff
@@ -1,58 +1,58 @@
...
+  \ {
+  \ 'lnum': 5, 'bufnr': bufnr('%'), 'col': 13, 'type': 'W',
+  \ 'text': "Missing semicolon.",
+  \ 'valid': 1, 'module': '', 'pattern': '', 'vcol': 0, 'nr': -1,
+  \ },
+  \ {
+  \ 'lnum': 7, 'bufnr': bufnr('%'), 'col': 2, 'type': 'E',
+  \ 'text': "Unnecessary semicolon.",
+  \ 'valid': 1, 'module': '', 'pattern': '', 'vcol': 0, 'nr': -1,
+  \ }]
```
Comment (blueyed) [inline tests/ft_javascript.vader:55]: Have you investigated the test failures there?
(I've only quickly seen that this failed with nvim-0.1.7)

**Review by blueyed — COMMENTED** [review]
This is good in general, thanks!

Tests need to be fixed.

The sasslint maker is using the errorformat that has been removed here, so should get adjusted / moved to there then: https://github.com/blueyed/neomake/blob/0a37942388174282fa5716a71e7b703795f2d63f/autoload/neomake/makers/ft/scss.vim#L11

**Comment by euclio** [comment]
Tests have been fixed.

**Review by blueyed — APPROVED** [review]

**Comment by blueyed** [comment]
Thank you!

**Pull request merged**